src/pushover_complete/pushover_api.py

```diff
@@ -136,10 +140,14 @@ def _send_message(self, user, message, device=None, title=None, url=None, url_ti
             'sound': sound,
             'html': html
         }
+        
+        file = {}
+        if image_path is not None and os.path.isfile(image_path):
+            file = {'attachment': (image_path, open(image_path, 'rb'))}
```

Comment:
this doesn't handle file cleanup. Would using a context manager and calling `read()` be an appropriate alternative?

Reply:
Good point. I whipped this together last night because I wanted to start pushing images. I didn't spend a ton of time thinking about the right way to do it.
I'll have to investigate though, Requests wants the file object and that would be cleaned up before the following call to post. It's probably enough to just put that call inside the with-open block.

Reply:
Maybe this, then?

```python
if image_path is not None and os.path.isfile(image_path):
    with open(image_path, 'rb') as f:
           file = {'attachment': (image_path, f)}
```

Reply:
That's what I was thinking:
```python
if image_path is not None and os.path.isfile(image_path):
    with open(image_path, 'rb') as f:
           file = {'attachment': (image_path, f)}
           return self._generic_post('messages.json', payload=payload, session=session, files=file)
```

Reply:
Moving the `open` call inside `_generic_post` will handle this issue as well.

Reply:
for now I still have the open call and dict construction inside `_send_message`. If you would prefer it move inside `_generic_post` I can do that. It seems to me to make more sense where it is, but I'm open.